backend/server/src/models/tournamentModel.ts

```diff
@@ -1,12 +1,39 @@
 import mongoose, { Schema, type Document, type Types } from "mongoose";
-import { type Match } from "./matchModel";
+import { type MatchPlayer, type Match } from "./matchModel";
 
 export enum TournamentType {
   RoundRobin = "Round Robin",
   Playoff = "Playoff",
   PreliminiaryPlayoff = "Preliminary Playoff"
 }
 
+export interface UnsavedMatch {
+  players: MatchPlayer[];
+  type: string;
+  admin: Types.ObjectId | null;
+  elapsedTime: number;
+  timerStartedTimestamp: Date | null;
+}
+
+interface PlayerDetails {
+  firstName: string | null;
+  lastName: string | null;
+  id: string | null;
+}
+
+export interface ExtendedMatch extends Match {
+  playersDetails: Array<{
+    id: string | null;
+    firstName: string | null;
+    lastName: string | null;
+  }>;
+  winnerDetails?: {
+    id: string | null;
+    firstName: string | null;
+    lastName: string | null;
+  } | null;
+}
```

Comment:
Do the properties need to be nullable as well considering that players are required to enter all of these information, therefore if the player exists, then these would exist as well?

Reply:
Also, use undefined over null if you must. Right now the winnderDetails has 3 types undefined | null | playerDetails

Reply:
Modified. 

```
export interface ExtendedMatch extends Match {
  playersDetails: Array<{
    id: Types.ObjectId;
    firstName: string;
    lastName: string;
  }>;
  winnerDetails?: PlayerDetails | null;
}
```